command-line-local-shortener/main.go

```diff
@@ -70,21 +90,26 @@ func getReq(c *gin.Context) {
 }
 
 func postReq(c *gin.Context) {
-	longURL := c.PostForm("longURL")
+	var link Shorten
+	c.BindJSON(&link)
 
-	if longURL == "" {
-		c.String(http.StatusBadRequest, "enter a url to shorten")
+	if link.LongURL == "" {
+		c.JSON(noURL.ErrCode, noURL.ErrMessage)
 		return
-	} else {
-		result, err := dataProccess(longURL)
-		if err != nil {
-			c.String(http.StatusBadRequest, "invalid URL")
-		} else {
-			c.String(http.StatusOK, fmt.Sprintf(
-				"Your new link is: %s/%s", SERVER_URL, result))
-		}
+	}
+	result, err := dataProccess(link.LongURL)
+	if err != nil {
+		c.JSON(invalidURL.ErrCode, invalidURL.ErrMessage)
+		return
+	}
+
+	response := Shorten{
+		LongURL: link.LongURL,
+		ShortURL: fmt.Sprintf(
```

Comment:
Just put `shortUrl` here. the front end will handle displaying a user-friendly message.